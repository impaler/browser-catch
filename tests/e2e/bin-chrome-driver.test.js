import test from 'ava'

require('babel-polyfill')
require('babel-register')

const fixtureServer = require('../../src/lib/fixture-server/server')
const ERRORS = require('../../src/lib/fixture-server/fixtures/errors')
const FEATURES = require('../../src/lib/fixture-server/fixtures/features')
const FIXTURES = [...ERRORS, ...FEATURES]
const fork = require('./../../src/lib/fork')
const {stripDefaultReporterHack} = require('./../helpers/result')

test('ChromeDriver catch 1 error', async t => {
  const serverSettings = await fixtureServer({routes: ERRORS})
  const errors = await fork(`${serverSettings.url}/error --driver chrome --json`)
  const errorsParsed = stripDefaultReporterHack(errors.stdout, serverSettings.port)

  t.is(errorsParsed.driverType, 'chrome')
  t.is(errorsParsed.errors.length, 1)
  t.snapshot(errorsParsed)
})

test('ChromeDriver catch 0 errors', async t => {
  const serverSettings = await fixtureServer({routes: ERRORS})
  const errors = await fork(`${serverSettings.url} --driver chrome`)
  const errorsParsed = stripDefaultReporterHack(errors.stdout, serverSettings.port)

  t.is(errorsParsed.driverType, 'chrome')
  t.is(errorsParsed.errors.length, 0)
  t.snapshot(errorsParsed)
})

test('ChromeDriver catch 0 errors to stdout & json', async t => {
  const serverSettings = await fixtureServer()
  const errors = await fork(`${serverSettings.url} --driver chrome --json`)
  const errorsParsed = stripDefaultReporterHack(errors.stdout, serverSettings.port)

  t.is(errorsParsed.driverType, 'chrome')
  t.is(errorsParsed.errors.length, 0)
  t.snapshot(errorsParsed)
})

test('ChromeDriver throws error when testing a url with a 404', async t => {
  const serverSettings = await fixtureServer({routes: ERRORS})
  const errors = await fork(`${serverSettings.url}/404 --driver chrome --json`)
  const errorsParsed = stripDefaultReporterHack(errors.stdout, serverSettings.port)

  t.is(errorsParsed.driverType, 'chrome')
  t.is(errorsParsed.errors.length, 1)
  t.snapshot(errorsParsed)
})

// Requires a custom selenium server url & port running
// Makes testing the rest in prallel difficult when the selenium/standalone-chrome
// fails with xvfb-run: error: Xvfb failed to start when more than one is running
// So it needs to be a separate test run
//
// test('webdriver-custom-url ChromeDriver catches 0 errors and works with custom selenium url', async t => {
//     const serverSettings = await fixtureServer({routes: ERRORS})
//     const errors = await fork(`${serverSettings.url}/404 --driver chrome --json --webdriver-port 3445 --webdriver-host localhost`)
//     const errorsParsed = stripDefaultReporterHack(errors.stdout, serverSettings.port)
//
//     t.is(errorsParsed.driverType, 'chrome')
//     t.is(errorsParsed.errors.length, 0)
//     t.snapshot(errorsParsed)
// })
