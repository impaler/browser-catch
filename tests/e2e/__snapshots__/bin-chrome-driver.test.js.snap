exports[`ChromeDriver catch 0 errors 1`] = `
Object {
  "driverType": "chrome",
  "errors": Array [],
}
`;

exports[`ChromeDriver catch 0 errors to stdout & json 1`] = `
Object {
  "driverType": "chrome",
  "errors": Array [],
}
`;

exports[`ChromeDriver catch 1 error 1`] = `
Object {
  "driverType": "chrome",
  "errors": Array [
    Object {
      "level": "SEVERE",
      "message": "http://127.0.1.1:9999/error 5:20 \"A regular console.error\"",
    },
  ],
}
`;

exports[`ChromeDriver throws error when testing a url with a 404 1`] = `
Object {
  "driverType": "chrome",
  "errors": Array [
    Object {
      "level": "SEVERE",
      "message": "http://127.0.1.1:9999/404 - Failed to load resource: the server responded with a status of 404 (Not Found)",
    },
  ],
}
`;
